src/main/java/cascading/tap/hcatalog/HCatTap.java

```diff
@@ -0,0 +1,173 @@
+   * @param scheme {@link HCatScheme} that describes the columns to be used.
+   * @param databaseName Name of the Hive database where {@code tableName} is defined.
+   * @param tableName Name of the Hive table that this {@link Tap} represents.
+   */
+  public HCatTap(HCatScheme scheme, String databaseName, String tableName) {
+    this(scheme, databaseName, tableName, null);
+  }
+
+  /**
+   * Constructs a new {@link HCatTap}.
+   * <p>
+   * This constructor should be use by source {@linkplain Tap Taps} when partition filtering is required. See <a href=
+   * "https://cwiki.apache.org/confluence/display/Hive/Hbase+execution+plans+for+RawStore+partition+filter+condition">
+   * HCatalog filter expressions</a> for details about the {@code filter} format.
+   * </p>
+   *
+   * @param scheme {@link HCatScheme} that describes the columns to be used.
+   * @param databaseName Name of the Hive database where {@code tableName} is defined.
+   * @param tableName Name of the Hive table that this {@link Tap} represents.
+   * @param filter A partition-selector expression.
+   */
+  public HCatTap(HCatScheme scheme, String databaseName, String tableName, String filter) {
+    super(scheme);
+    this.databaseName = databaseName;
+    this.tableName = tableName;
+    this.filter = filter;
+  }
+
+  @Override
+  public String getIdentifier() {
+    return String.format("hcatalog://%s.%s", databaseName, tableName);
+  }
+
+  @Override
+  public void sourceConfInit(FlowProcess<? extends Configuration> flowProcess, Configuration conf) {
+    DeprecatedInputFormatWrapper.setInputFormat(HCatInputFormat.class, (JobConf) conf,
+        HcatInputFormatValueCopier.class);
+    try {
+      HCatInputFormat.setInput(conf, databaseName, tableName, filter);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    super.sourceConfInit(flowProcess, conf);
+  }
+
+  @Override
+  public TupleEntryIterator openForRead(FlowProcess<? extends Configuration> flowProcess, RecordReader input)
+    throws IOException {
+    return new HadoopTupleEntrySchemeIterator(flowProcess, this, input);
+  }
+
+  @Override
+  public void sinkConfInit(FlowProcess<? extends Configuration> flowProcess, Configuration conf) {
+    DeprecatedOutputFormatWrapper.setOutputFormat(HCatOutputFormat.class, (JobConf) conf);
+    OutputJobInfo outputJobInfo = OutputJobInfo.create(databaseName, tableName, null);
+    try {
+      HCatOutputFormat.setOutput(conf, ((JobConf) conf).getCredentials(), outputJobInfo);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    super.sinkConfInit(flowProcess, conf);
+  }
+
+  @Override
+  public TupleEntryCollector openForWrite(FlowProcess<? extends Configuration> flowProcess, OutputCollector output)
+    throws IOException {
+    return new HadoopTupleEntrySchemeCollector(flowProcess, this, output);
+  }
+
+  @Override
+  public boolean resourceExists(Configuration conf) throws IOException {
+    return true;
+  }
```

Comment:
These need to be better defined, otherwise they might cause unexpected behaviour esp. when things run in a Cascade